go/login/server/server.go

```diff
@@ -0,0 +1,178 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package server implements GLOME-login server framework.
+package server
+
+import (
+	"encoding/hex"
+	"fmt"
+	"net/http"
+	"sync"
+
+	"../../login" // Modify when it is pushed
+)
+
+const (
+	// MaxResponseSize is the maximum size in characaters of the response token
+	MaxResponseSize = 44 // 32 bytes base64 encoded
+)
+
+var (
+	// ErrInvalidResponseLen denotes that response length provided is invalid
+	ErrInvalidResponseLen = fmt.Errorf("invalid response length provided")
+)
+
+// Authorizer responds to an authorization request. The method
+// GrantLogin returns whether an user is allowed to run an action in a host, and a error value.
+//
+// Some considerations need to be held while implementing this interface:
+// - Allow should accept an empty string as command, denoting that
+// an empty action is passed in the request.
+// - If no user can be obtained from request metadata, an empty string is to be
+// passed as default value.
+// - Both hostIDType and hostID can be empty. Whether this refer to a default value
+// or not is to be user configurable.
```

Comment:
Clarify that the returned boolean will be considered even if an error is returned, so that nobody thinks they can get away with laziness.

Reply:
Added! even so, zero value for boolean in Golang is false, which is probably the value you want if you are to return an Error so the might get away with laziness afterall :)